qualtran/bloqs/arithmetic/negate_test.py

```diff
@@ -0,0 +1,44 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import pytest
+
+from qualtran import QInt, QUInt
+from qualtran.bloqs.arithmetic.negate import _negate, _negate_symb, Negate
+
+
+def test_examples(bloq_autotester):
+    bloq_autotester(_negate)
+    bloq_autotester(_negate_symb)
+
+
+@pytest.mark.parametrize("bitsize", [1, 2, 3, 4, 8])
+def test_negate_classical_sim(bitsize: int):
+    # TODO use QInt once classical sim is fixed.
+    #      classical sim currently only supports unsigned.
```

Comment:
Is this true? I thought classical simulation supports both signed and unsigned integers. Is there an open issue you are referring to?

Reply:
https://github.com/quantumlib/Qualtran/issues/606

Reply:
Thanks, looks like it's bottlenecked by speed and https://github.com/quantumlib/Qualtran/pull/1142/ should help fix it?